test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,352 @@
+)
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	const (
+		agnHostLabelKey      = "app"
+		agnHostLabelValue    = "agnhost"
+		agnHostPodNamePreFix = "agnhost-"
+
+		dnsPort              = 53
+		customDnsServiceName = "custom-dns-service"
+	)
+	skipIfAntreaPolicyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	customDnsService, err := data.CreateServiceWithAnnotations(customDnsServiceName, data.testNamespace, dnsPort, dnsPort, corev1.ProtocolUDP,
+		map[string]string{customDnsLabelKey: customDnsLabelValue}, false, false, corev1.ServiceTypeClusterIP, ptr.To[corev1.IPFamily](corev1.IPv4Protocol), map[string]string{})
+	require.NoError(t, err, "Error when creating custom DNS service: %v", err)
+
+	setDnsServerAddressInAntrea(t, data, customDnsService.Spec.ClusterIP)
+	defer setDnsServerAddressInAntrea(t, data, "")
+
+	podCount := 2
+	agnHostPodIps := make([]*PodIPs, podCount)
+	for i := 0; i < podCount; i++ {
+		agnHostPodIps[i] = createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+	}
+
+	ipv4, ipv6 := extractIPs(agnHostPodIps[0])
+
+	dnsConfigData := createDnsConfig(t, ipv4, ipv6, testFullyQualifiedDomainName)
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      customDnsConfigName,
+			Namespace: data.testNamespace,
+		},
+		Data: dnsConfigData,
+	}
+	customDnsConfigMapObject, err := data.CreateConfigMap(configMap)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createDnsPod(t, data)
+
+	k8sUtils, err = NewKubernetesUtils(data)
+	require.NoError(t, err, "error getting k8s utils %+v", err)
+
+	createFqdnPolicyInNamespace(t, data, testFullyQualifiedDomainName)
+
+	t.Logf("Creating Toolbox pod...")
+	createToolBoxPod(t, data, customDnsService.Spec.ClusterIP)
+
+	curlTarget := func(podName, containerName, fqdn string) (string, error) {
+		cmd := []string{"curl", fqdn}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, podName, containerName, cmd)
+		if err != nil {
+			return "", fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), podName, err, stdout, stderr)
+		}
+		return stdout, nil
+	}
+
+	t.Logf("Trying to curl FQDN ...")
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		stdout, err := curlTarget(toolboxPodName, toolboxContainerName, testFullyQualifiedDomainName)
+		assert.NoError(t, err)
+		t.Logf("response of curl to FQDN - %+v", stdout)
+	}, 2*time.Second, 100*time.Millisecond, "trying to curl the fqdn : ", testFullyQualifiedDomainName)
+
+	t.Logf("Trying to DIG the FQDN to simulate caching the current IP inside Toolbox pod...")
+	digResponse, err := k8sUtils.runDNSQuery(toolboxPodName, toolboxContainerName, data.testNamespace, testFullyQualifiedDomainName, false, customDnsService.Spec.ClusterIP)
```

Comment:
`digResponse` is a poor variable name here. The fact that `dig` is used as an implementation detail of `runDNSQuery` is irrelevant to this test function.